contracts/validators/CredentialAtomicQueryMTPValidator.sol

```diff
@@ -28,14 +41,62 @@ contract CredentialAtomicQueryMTPValidator is CredentialAtomicQueryValidator {
         super.initialize(_verifierContractAddr, _stateContractAddr);
     }
 
-    function _getInputValidationParameters(
+    function verify(
+        uint256[] calldata inputs,
+        uint256[2] calldata a,
+        uint256[2][2] calldata b,
+        uint256[2] calldata c,
+        bytes calldata data
+    ) external view virtual returns (bool) {
+        CredentialAtomicQuery memory credAtomicQuery = abi.decode(data, (CredentialAtomicQuery));
+        IVerifier verifier = _circuitIdToVerifier[credAtomicQuery.circuitIds[0]];
+
+        require(
+            credAtomicQuery.circuitIds.length == 1 && verifier != IVerifier(address(0)),
+            "Invalid circuit ID"
+        );
+
+        // verify that zkp is valid
+        require(verifier.verify(a, b, c, inputs), "Proof is not valid");
+
+        // parse public signals from inputs array
+        PublicSignals memory signals = parsePublicSignals(inputs);
+
+        // check circuitQueryHash
+        require(
+            signals.circuitQueryHash == credAtomicQuery.queryHash,
+            "Query hash does not match the requested one"
+        );
+
+        // TODO: add support for query to specific userID and then verifying it
+
+        // check isRevocationChecked flag
+        // TODO: add support for skipping revocation check to query
+        // until then it should be verified as enabled
+        require(signals.isRevocationChecked == 1, "Revocation check should be enabled");
```

Comment:
if we have setup skipClaimRevocationCheck  flag we should fail if the revocation check is not expected from user.